feeadjuster/feeadjuster.py

```diff
@@ -0,0 +1,136 @@
+        # some pseudo-randomness to avoid too easy channel balance probing
+        update_treshold = plugin.update_treshold
+        if not plugin.deactivate_fuzz:
+            update_treshold += random.uniform(-0.015, 0.015)
+
+        if (last_percentage is None
+                or abs(last_percentage - percentage) > update_treshold):
+            ratio = get_ratio(percentage)
+            try:
+                plugin.rpc.setchannelfee(scid, int(plugin.adj_basefee * ratio),
+                                         int(plugin.adj_ppmfee * ratio))
+                plugin.log("Adjusted fees of {} with a ratio of {}"
+                           .format(scid, ratio))
+                plugin.adj_balances[scid]["last_percentage"] = percentage
+            except RpcError as e:
+                plugin.log("Could not adjust fees for channel {}: '{}'"
+                           .format(scid, e), level="warn")
+
+
+def get_chan(plugin: Plugin, scid: str):
+    for peer in plugin.rpc.listpeers()["peers"]:
+        if len(peer["channels"]) == 0:
+            continue
+        # We might have multiple channel entries ! Eg if one was just closed
+        # and reopened.
+        for chan in peer["channels"]:
+            if "short_channel_id" not in chan:
+                continue
+            if chan["short_channel_id"] == scid:
+                return chan
+
+
+def maybe_add_new_balances(plugin: Plugin, scids: list):
+    for scid in scids:
+        if scid not in plugin.adj_balances:
+            # At this point we could call listchannels and pass the scid as
+            # argument, and deduce the peer id (!our_id). But -as unlikely as
+            # it is- we may not find it in our gossip (eg some corruption of
+            # the gossip_store occured just before the forwarding event).
+            # However, it MUST be present in a listpeers() entry.
+            chan = get_chan(plugin, scid)
+            assert chan is not None
+
+            plugin.adj_balances[scid] = {
+                "our": int(chan["to_us_msat"]),
+                "total": int(chan["total_msat"])
+            }
+
+
+@plugin.subscribe("forward_event")
+def forward_event(plugin: Plugin, forward_event: dict, **kwargs):
+    if forward_event["status"] == "settled":
+        in_scid = forward_event["in_channel"]
+        out_scid = forward_event["out_channel"]
+        maybe_add_new_balances(plugin, [in_scid, out_scid])
+
+        plugin.adj_balances[in_scid]["our"] += forward_event["in_msatoshi"]
+        plugin.adj_balances[out_scid]["our"] -= forward_event["out_msatoshi"]
+        try:
+            # Pseudo-randomly add some hysterisis to the update
+            if not plugin.deactivate_fuzz and random.randint(0, 9) == 9:
+                time.sleep(random.randint(0, 5))
+            maybe_adjust_fees(plugin, [in_scid, out_scid])
+        except Exception as e:
+            plugin.log("Adjusting fees: " + str(e), level="error")
```

Comment:
```suggestion
            plugin.log("Adjusting fees: {e}".format(e=e), level="error")
```
`format` can be more efficient than string concatenation (might not be in this case with just 1 placeholder but it's a good habit to have).

Reply:
Think i can even use fstrings now ? :D

Reply:
Oh that's true, way better :+1: